lib/xclarity_client/xclarity_management_mixin.rb

```diff
@@ -6,17 +6,17 @@ def get_all_resources (resource, opts = {})
       response = connection(resource::BASE_URI, opts)
 
       $lxca_log.info "XclarityClient::ManagementMixin get_all_resources", "Response received from #{resource::BASE_URI}"
-      
+
       return [] unless response.success?
 
       body = JSON.parse(response.body)
       if resource == XClarityClient::User
         body = body['response']
       end
 
-      body = {resource::LIST_NAME => body} if body.is_a? Array
-      body = {resource::LIST_NAME => [body]} unless body.has_key? resource::LIST_NAME
-      body[resource::LIST_NAME].map do |resource_params|
+      list_name, body = add_listname_on_body(resource, body)
```

Comment:
You should add tests for this new approach(using `add_listname_on_body`) instead of the previous approach.

Reply:
Acctually this method with this new approach is already tested, but in a transitive way, this because this method is defined on a mixin that is used for a couple of classes and this classes that includes this mixin already test the flow that this method it's part.

If we take a look on the [spec file of the scalable_complex](https://github.com/lenovo/xclarity_client/blob/master/spec/xclarity_client_scalable_complex_spec.rb), for exemple, we will see this method, with the new approach, being called on the line 34:

```
it 'should respond with an array' do
    expect(@client.discover_scalableComplexes.class).to eq(Array)
end
```
Just to clarify, the call stack looks like this:
```
XClarityClient::Client#discover_scalableComplexes
  - ScalableComplexManagement#population
    - ManagementMixin#get_all_resources
```
And if we take all the resources specs there is an structure very similar with the described above.

So, I believe that our tests covers well this flow. Don't you think?

Reply:
Okay, now it makes sense for me! Sorry about take this long discussion! 😅